src/ui/component/collapsible/collapsible.tsx

```diff
@@ -0,0 +1,52 @@
+import type { FC } from 'react'
+import { useState, useCallback } from 'react'
+import classNames from 'classnames'
+import * as RCollapsible from '@radix-ui/react-collapsible'
+import type { IconName } from '@/ui/component/icon/icon'
+import { Icon } from '@/ui/component/icon/icon'
+import type { AssertSubset } from '@/util/type'
+import './collapsible.scss'
+
+type CollapsibleProps = {
+  children: JSX.Element
+  triggerElement: JSX.Element
+  open?: boolean
+  triggerClassName?: string
+  chevronIconName?: AssertSubset<IconName, 'chevron' | 'chevron-sharp'>
```

Comment:
I think we could let the user choose his own icon to represent the opening/closing of this component.
What's more, if we ever change our mind about which icon to use in our app, it forces us to modify the places where we use this component + modify the props of the component itself. I'm in favor of leaving the choice of icon free, with a default name to simplify its use.

Reply:
Thanks for your suggestion! It brings in fact more flexibility and scalability to the component. In this case, if the user doesn't want any icon, he must explicitly pass a value in order to avoid the fallback on the default parameter, `chevron`.

Is it OK to use `null` as a way to not display any icon, like this:
```suggestion
  iconName?: IconName | null
```

Reply:
I would leave the chevron icon as default, and the user can choose a different one if he prefers.
So I won't add `null` here...